src/components/tl/ucp/tl_ucp_sendrecv.h

```diff
@@ -151,6 +151,203 @@ static inline ucc_status_t ucc_tl_ucp_send_nz(void *buffer, size_t msglen,
+{
+    ucp_request_param_t req_param = {0};
+    ucs_status_ptr_t    ucp_status;
+    ucc_status_t        status;
+    ucp_ep_h            ep;
+    ucp_rkey_h          rkey;
+    uint64_t            rva;
+    int                 segment = 0;
+
+    // get the endpoint or create if doesn't exist
+    status = ucc_tl_ucp_get_ep(team, dest_group_rank, &ep);
+    if (ucc_unlikely(UCC_OK != status)) {
+        return status;
+    }
+
+    // resolve the p2p info
+    status = ucc_tl_ucp_resolve_p2p_by_va(team, target, &ep, dest_group_rank,
+                                          &rva, &rkey, &segment);
+    if (ucc_unlikely(UCC_OK != status)) {
+        return status;
+    }
+
+    rva = rva + ((ptrdiff_t)target - (ptrdiff_t)team->va_base[segment]);
+
+    // issue operation
+    ucp_status = ucp_put_nbx(ep, buffer, msglen, rva, rkey, &req_param);
+
+    if (UCC_OK != ucp_status) {
+        ucp_request_free(ucp_status);
+    }
+    return UCC_OK;
+}
+
+static inline ucc_status_t ucc_tl_ucp_get_nb(void *buffer, void *target,
+                                             size_t             msglen,
+                                             ucc_rank_t         dest_group_rank,
+                                             ucc_tl_ucp_team_t *team,
+                                             ucc_tl_ucp_task_t *task)
+{
+    ucp_request_param_t req_param = {0};
+    ucs_status_ptr_t    ucp_status;
+    ucc_status_t        status;
+    ucp_ep_h            ep;
+    ucp_rkey_h          rkey;
+    uint64_t            rva;
+    int                 segment = 0;
+
+    // get the endpoint or create if doesn't exist
+    status = ucc_tl_ucp_get_ep(team, dest_group_rank, &ep);
+    if (ucc_unlikely(UCC_OK != status)) {
+        return status;
+    }
+
+    // resolve the p2p info
+    status = ucc_tl_ucp_resolve_p2p_by_va(team, target, &ep, dest_group_rank,
+                                          &rva, &rkey, &segment);
+    if (ucc_unlikely(UCC_OK != status)) {
+        return status;
+    }
+    rva = rva + ((ptrdiff_t)target - (ptrdiff_t)team->va_base[segment]);
+
+    // issue operation
+    ucp_status = ucp_get_nbx(ep, buffer, msglen, rva, rkey, &req_param);
+
+    if (UCC_OK != ucp_status) {
+        if (UCS_PTR_IS_ERR(ucp_status)) {
+            return UCC_ERR_NO_MESSAGE;
+        }
+        ucp_request_free(ucp_status);
+    }
+
+    return UCC_OK;
+}
+
+static inline ucc_status_t ucc_tl_ucp_atomic_inc(void *     target,
+                                                 ucc_rank_t dest_group_rank,
+                                                 ucc_tl_ucp_team_t *team,
+                                                 ucc_tl_ucp_task_t *task)
+{
+    ucp_request_param_t req_param = {0};
+    ucs_status_ptr_t    ucp_status;
+    ucc_status_t        status;
+    ucp_ep_h            ep;
+    ucp_rkey_h          rkey;
+    uint64_t            rva;
+    int                 segment = 0;
```

Comment:
following ucx codestyle initialized variables should be decaled first, more info https://github.com/openucx/ucx/blob/master/docs/CodeStyle.md